### PR TITLE
[rush] Replace most os.EOL references in rush-lib with '\n' literals.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: common/changes/@microsoft/rush/remove-os-eol_2022-12-05-08-38.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "",
+      "type": "none",
```
Comment (octogonz) [inline common/changes/@microsoft/rush/remove-os-eol_2022-12-05-08-38.json:5]: This seems like a user-impacting change, for example if someone has Jest snapshots that diff Rush output.

File: libraries/rush-lib/src/api/ApprovedPackagesConfiguration.ts [hunk 2]
```diff
@@ -2,7 +2,6 @@
 // See LICENSE in the project root for license information.
 
 import * as path from 'path';
```
Comment (octogonz) [inline libraries/rush-lib/src/api/ApprovedPackagesConfiguration.ts:4]: We had some discussion about whether `console.log()` would produce CRLF on Windows.  Empirical tests confirmed that it produces LF, and @iclanton  later confirmed this in the [Node.js source code](https://github.com/nodejs/node/blob/main/lib/internal/console/constructor.js#L287).

Thus it does seem that this PR should mostly eliminate CRLF from Rush's output. 👍

**Review by octogonz — APPROVED** [review]

File: libraries/rush-lib/src/api/ApprovedPackagesConfiguration.ts [hunk 3]
```diff
@@ -2,7 +2,6 @@
 // See LICENSE in the project root for license information.
```
[the remark this answers is not in the document]
Comment (iclanton) [reply libraries/rush-lib/src/api/ApprovedPackagesConfiguration.ts:2]: I think we should, for logging at least.

File: common/changes/@microsoft/rush/remove-os-eol_2022-12-05-08-38.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/remove-os-eol_2022-12-05-08-38.json:4]:
```suggestion
      "comment": "Normalize all newlines in logging statements to LF. On Windows, some newlines were CRLF.",
```

**Pull request merged**